cmd/collectors/zapi/plugins/volume/volume.go

```diff
@@ -92,19 +92,25 @@ func (v *Volume) Run(dataMap map[string]*matrix.Matrix) ([]*matrix.Matrix, error
 	}
 
 	volumeCloneMap, err := v.getVolumeCloneInfo()
-
 	if err != nil {
 		v.Logger.Error().Err(err).Msg("Failed to update clone data")
 	}
 
+	volumeFootprintMap, err := v.getVolumeFootprint()
+	if err != nil {
+		v.Logger.Error().Err(err).Msg("Failed to update footprint data")
+		// clean the map in case of the error
+		volumeFootprintMap = make(map[string]map[string]string)
```

Comment:
`clear(volumeFootprintMap)` unless you need a new instance

Reply:
Earlier I tried clear(), but it was before the initialising, which gives NPE. This would be right place for clear().
Added